parsers/US_ERCOT.py

```diff
@@ -374,3 +392,140 @@ def fetch_exchange(
     return exchanges
+
+
+def _get_dataframe_from_url(url, session):
+    response = session.get(url)
+    docs = response.json()["ListDocsByRptTypeRes"]["DocumentList"]
+
+    # Initialize variables to track the latest PublishDate and corresponding DocID
+    latest_publish_date = None
+    latest_doc_id = None
+
+    # Iterate through the list of documents
+    for item in docs:
+        publish_date_str = item["Document"]["PublishDate"]
+        publish_date = datetime.fromisoformat(publish_date_str)
+
+        # Compare with the current latest_publish_date
+        if latest_publish_date is None or publish_date > latest_publish_date:
+            latest_publish_date = publish_date
+            latest_doc_id = item["Document"]["DocID"]
+
+    doc_url = f"{US_PROXY}/misdownload/servlets/mirDownload?doclookupId={latest_doc_id}&{HOST_PARAMETER}"
+    resp: Response = session.get(doc_url)  # verify=False
+
+    # Open the ZIP file
+    with zipfile.ZipFile(BytesIO(resp.content)) as z:
+        # Extract the first file in the ZIP (assuming it contains a CSV)
+        csv_filename = z.namelist()[0]
+        with z.open(csv_filename) as f:
+            df = pd.read_csv(f)
+
+    return df
+
+
+def fetch_consumption_forecast(
+    zone_key: ZoneKey = ZoneKey("US-TEX-ERCO"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    """Requests load forecast data in MW. target_datetime not implemented (takes the latest report)"""
+    session = session or Session()
+
+    url = f"{US_PROXY}/misapp/servlets/IceDocListJsonWS?reportTypeId={LOAD_FORECAST_BY_FORECAST_ZONE}&_{int(time.time())}&{HOST_PARAMETER}"
+    df = _get_dataframe_from_url(url, session)
+
+    # Transfrom Hour column
+    df["HourStarting"] = df["HourEnding"].str.split(":", expand=True)[0].astype(int) - 1
+    df["HourStarting"] = df["HourStarting"].astype(str) + ":00"
+
+    all_consumption_events = (
+        df.copy()
+    )  # all events with a datetime and a consumption value
+    consumption_list = TotalConsumptionList(logger)
+    for _, row in all_consumption_events.iterrows():
+        date_, time_ = row.DeliveryDate, row.HourStarting
+        datetime_object = datetime.strptime(
+            date_ + "T" + time_, "%m/%d/%YT%H:%M"
+        ).replace(tzinfo=TX_TZ)
+        consumption_list.append(
+            zoneKey=zone_key,
+            datetime=datetime_object,
+            consumption=row.SystemTotal,
+            source=SOURCE,
+            sourceType=EventSourceType.forecasted,
+        )
+    return consumption_list.to_list()
+
+
+def fetch_wind_solar_forecasts(
+    zone_key: ZoneKey = ZoneKey("US-TEX-ERCO"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    """Requests wind and solar power data in MW. target_datetime not implemented (takes the latest report)"""
```

Comment:
Does it support it at all?

It would be good to have so we can backfill all the data to use more advanced estimation methods.